numpy/lib/function_base.py

```diff
@@ -3706,7 +3706,8 @@ def percentile(a, q, axis=None, out=None,
         plt.show()
 
     """
-    q = np.true_divide(q, 100.0)  # handles the asarray for us too
+    q = np.true_divide(q, 100)
+    q = asanyarray(q)  # undo any decay that the ufunc performed (see gh-13105)
```

Comment:
Is the asanyarray call an additional bugfix or was it somehow done by the use of float(100) previously? If it is a bugfix, is there a test that failed before and passes now?

Reply:
This is a bugfix for all object arrays - previously calling with `np.array(1, dtype=object)` would fail with an `AttributeError`. This is covered by the `Fraction` tests now.

Reply:
Strange no-one reported it until now. I guess percentile/quantile of object arrays is not a popular use case